### DISCUSSION TITLE
(Quit on unsupport PostgreSQL version) Database connection error Retrying the connection

### DISCUSSION THREAD

**Discussion by sseenivasan89** [PostgREST/postgrest#2904] [Q&A] [answered]
### Environment

* PostgreSQL version: 7.4 (not in docker)
* PostgREST version: 11.2.0 (c820efb) 
* Operating system: Windows 11

### Description of the issue
getting database connection error, shared my postgrest.conf file as well

I installed PostgresDB and install postgrest and setup postgrest.conf and tried to connect 

![image](https://github.com/PostgREST/postgrest/assets/44080743/1d3631b2-8fd2-4481-83ac-9fbb61e8bc40)

# postgrest.conf
db-uri = "postgres://postgres:postgres@host:5432/my_db"
db-anon-role = "anon"
db-schema="public"
server-port = 3000

**Comment by steve-chavez** [comment]
> PostgreSQL version: 7.4

We only support PostgreSQL 9.6 and up, see https://postgrest.org/en/stable/explanations/install.html#supported-postgresql-versions.

PostgreSQL only started supporting JSON functions since 9.3, so 7.4 is unfeasible to support.

---

I'm surprised we're not responding with an error though, we have:

https://github.com/PostgREST/postgrest/blob/87d6a0d0feaab6a85aaa5ba7ec6c0e9f97855fe4/src/PostgREST/AppState.hs#L337-L343

We could bring back pg 9.5 (https://github.com/PostgREST/postgrest/pull/2038) to be sure we test the above.

**Comment by sseenivasan89** [comment]
@steve-chavez 

Sorry my bad, i am using 15.3

![image](https://github.com/PostgREST/postgrest/assets/44080743/b02596c4-5f9b-49c7-bad2-15a12f587e97)

**Reply by steve-chavez** [reply, marked as answer]
As per your error message, `host` seems incorrect below:

> db-uri = "postgres://postgres:postgres@host:5432/my_db"

I suggest to try and connect with regular `psql`, then use the same connection string for PostgREST.

**Reply by sseenivasan89** [reply]
Hi @steve-chavez 

Thanks, my bad i update the `host` as invalid one 
I am able to connect the server, Thanks !
